### PR TITLE
rework repos and get months

### PR DESCRIPTION
**Pull request by BelehovEgor**
rework repositories to async
now we can decide how many next month will be taken
add links to position and department services

### CODE REVIEW

**Review by Cheephot — APPROVED** [review]

File: src/TimeService.Business/Commands/LeaveTime/FindLeaveTimesCommand.cs [hunk 1]
```diff
@@ -69,7 +79,7 @@ private async Task<List<UserData>> GetUsersDataFromBroker(List<Guid> usersIds, L
 
       try
       {
-        Response<IOperationResult<IGetUsersDataResponse>> response = await _rcGetUsers.GetResponse<IOperationResult<IGetUsersDataResponse>>(
+        var response = await _rcGetUsers.GetResponse<IOperationResult<IGetUsersDataResponse>>(
```
Comment (MariyaSin) [inline src/TimeService.Business/Commands/LeaveTime/FindLeaveTimesCommand.cs:82]: why do you use var here?

File: src/TimeService.Business/Commands/Stat/FindStatCommand.cs [hunk 2]
```diff
@@ -103,6 +109,59 @@ private async Task<List<ProjectData>> GetProjectsThroughBroker(List<Guid> projec
...
+    }
+
+    private async Task<List<UserData>> GetUsersDataFromBrokerAsync(List<Guid> usersIds, List<string> errors)
+    {
+      if (usersIds == null || !usersIds.Any())
+      {
+        return null;
+      }
+
+      string message = "Cannot get users data. Please try again later.";
```
Comment (MariyaSin) [inline src/TimeService.Business/Commands/Stat/FindStatCommand.cs:138]: What for do you create the message here, If you use it only 160 string?

**Review by MariyaSin — APPROVED** [review]

File: src/TimeService.Business/Commands/LeaveTime/FindLeaveTimesCommand.cs (hunk 1 above)
Comment (BelehovEgor) [reply src/TimeService.Business/Commands/LeaveTime/FindLeaveTimesCommand.cs:82]: fixed

File: src/TimeService.Business/Commands/Stat/FindStatCommand.cs (hunk 2 above)
Comment (BelehovEgor) [reply src/TimeService.Business/Commands/Stat/FindStatCommand.cs:138]: fixed

**Review by VLiamin — APPROVED** [review]

**Pull request merged**